vehicle_status_generator/src/vehicle_status_generator.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2019-2021 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+#include "vehicle_status_generator.h"
+
+namespace vehicle_status_generator
+{
+    VehicleStatusGenerator::VehicleStatusGenerator() : vehicle_status_generation_frequency_(10.0) {}
+
+    void VehicleStatusGenerator::initialize()
+    {
+        nh_.reset(new ros::CARMANodeHandle());
+        pnh_.reset(new ros::CARMANodeHandle("~"));
+        
+        pnh_->param<double>("vehicle_status_generation_frequency_", vehicle_status_generation_frequency_, 10.0);
+        nh_->param<double>("vehicle_length", vehicle_length_, 5.0);
+        nh_->param<double>("vehicle_width", vehicle_width_, 2.0);
+
+        nh_->param<double>("/vehicle_acceleration_limit", vehicle_acceleration_limit_, 2.0);
+        nh_->param<double>("/vehicle_deceleration_limit", vehicle_deceleration_limit, 2.0);
+        nh_->param<double>("/guidance/yield_plugin/x_gap", x_gap, 2.0);
```

Comment:
I think at least this param would be better served being read from `x_gap`, and then remapped to `/guidance/yield_plugin/x_gap` in the launch file. The others are okay as they're more general, vehicle-wide parameters, but this is reading another plugins plugin-specific config so we should handle it with a bit more abstraction.